connector/connect/server/src/test/scala/org/apache/spark/sql/connect/service/SparkConnectServiceE2ESuite.scala

```diff
@@ -202,4 +202,17 @@ class SparkConnectServiceE2ESuite extends SparkConnectServerTest {
       }
     }
   }
+
+  test("SPARK-47144: Verify collations work with Connect client") {
+    withClient { client =>
+      Seq("UCS_BASIC", "UCS_BASIC_LCASE", "UNICODE", "UNICODE_CI").foreach(collationName => {
+        val query1 = client.execute(buildPlan(s"select 'abc' collate '$collationName'"))
+        query1.hasNext // guarantees the request was received by server.
+
+        val query2 = client.execute(buildPlan(s"select collation('abc' collate '$collationName')"))
+        query2.hasNext // guarantees the request was received by server.
+      })
+      client.releaseSession()
```

Comment:
Looks like the purpose of your test here is to make sure the request is passing through without an error. You do not need this test. 

What you need is to add a test in `PlanGenerationTestSuite` as your purpose is to verify the client generated a correct plan. You can check the class doc of `PlanGenerationTestSuite` to understand how to add such a test.

Reply:
@amaliujia Added test to `PlanGenerationTestSuite`, please check. There were also some changes to golden files related to adding `collationId` in string types.

Reply:
It is expected to have the changes in the golden files.